### ISSUE TITLE
Using with rsconnect to deploy shinyApp

### ISSUE THREAD

**Issue by alptaciroglu** [cytoscape/cyjShiny#16]
I would like to deploy my shiny app, which uses this package. This package falls into the category of private packages (I think) described in the link: 

https://support.rstudio.com/hc/en-us/articles/226871467-Package-management-in-RStudio-Connect

After several failed attempts on how to apply the support document instructions on cyjShiny, I wanted to ask if anyone here has encountered and know a solution to this. 

Running on Ubuntu18.04 and R 3.5.1

**Comment by paul-shannon** [comment]
I am the author and maintainer of cyjShiny.  Your bug report is most welcome - I will do what I can to help.

Can you provide a detailed error report - sufficiently detailed so that I can reproduce the difficulty you encountered?

 - Paul

**Comment by alptaciroglu** [comment]
Hey Paul, thanks for your answer.

Thanks in advance.

Here is what I get:

> library(rsconnect)
> rsconnect::deployApp(account='xxxxxx', appDir=getwd(), appFiles=list.files(), appTitle='Odyssey', appName='Odyssey')
Preparing to deploy application...Update application currently deployed at
https://diagnose.shinyapps.io/odyssey/? [Y/n] Y
DONE
Uploading bundle for application: 310926...DONE
Deploying bundle: 1876337 for application: 310926 ...
Waiting for task: 587465137
  building: Processing bundle: 1876337
  building: Parsing manifest
  building: Building image: 1953587
  building: Installing system dependencies
  building: Fetching packages
  error: Building package: rcytoscapejs
################################ Begin Task Log ################################ 
[2019-02-14T15:47:36.378407613+0000] Execute script: packages/build/sys.sh
+ > 
set -e
+ > 
apt-get update -qq
+ > 
apt-get install -y libapparmor-dev
Reading package lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  libapparmor-dev
0 upgraded, 1 newly installed, 0 to remove and 14 not upgraded.
Need to get 67.5 kB of archives.
After this operation, 228 kB of additional disk space will be used.
Get:1 http://us-east-1.ec2.archive.ubuntu.com/ubuntu xenial-updates/main amd64 libapparmor-dev amd64 2.10.95-0ubuntu2.10 [67.5 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 67.5 kB in 0s (0 B/s)
Selecting previously unselected package libapparmor-dev:amd64.
(Reading database ... 119344 files and directories currently installed.)
Preparing to unpack .../libapparmor-dev_2.10.95-0ubuntu2.10_amd64.deb ...
Unpacking libapparmor-dev:amd64 (2.10.95-0ubuntu2.10) ...
Processing triggers for man-db (2.7.5-1) ...
Setting up libapparmor-dev:amd64 (2.10.95-0ubuntu2.10) ...
################################# End Task Log ################################# 
Error: Unhandled Exception: Child Task 587465138 failed: Error building image: Error fetching rcytoscapejs (0.0.7) source. <CRANPackageSource repo='http://cran.rstudio.org'> unable to satisfy package: rcytoscapejs (0.0.7)
In addition: Warning message:
In FUN(X[[i]], ...) :
  Package 'rcytoscapejs 0.0.7' was installed from sources; Packrat will assume this package is available from a CRAN-like repository during future restores
> 

##################################################################################

─ Session info ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.5.2 (2018-12-20)
 os       Ubuntu 18.04.1 LTS          
 system   x86_64, linux-gnu           
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Europe/Istanbul             
 date     2019-02-14                  

─ Packages ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 package      * version   date       lib source                                     
 askpass        1.1       2019-01-13 [1] CRAN (R 3.5.2)                             
 assertthat     0.2.0     2017-04-11 [1] CRAN (R 3.5.2)                             
 backports      1.1.3     2018-12-14 [1] CRAN (R 3.5.2)                             
 bindr          0.1.1     2018-03-13 [1] CRAN (R 3.5.2)                             
 bindrcpp       0.2.2     2018-03-29 [1] CRAN (R 3.5.2)                             
 bitops         1.0-6     2013-08-17 [1] CRAN (R 3.5.2)                             
 callr          3.1.1     2018-12-21 [1] CRAN (R 3.5.2)                             
 cli            1.0.1     2018-09-25 [1] CRAN (R 3.5.2)                             
 codetools      0.2-16    2018-12-24 [4] CRAN (R 3.5.2)                             
 colorspace     1.4-0     2019-01-13 [1] CRAN (R 3.5.2)                             
 crayon         1.3.4     2017-09-16 [1] CRAN (R 3.5.2)                             
 crosstalk      1.0.0     2016-12-21 [1] CRAN (R 3.5.2)                             
 curl           3.3       2019-01-10 [1] CRAN (R 3.5.2)                             
 data.table   * 1.12.0    2019-01-13 [1] CRAN (R 3.5.2)                             
 desc           1.2.0     2018-05-01 [1] CRAN (R 3.5.2)                             
 devtools     * 2.0.1     2018-10-26 [1] CRAN (R 3.5.2)                             
 digest         0.6.18    2018-10-10 [1] CRAN (R 3.5.2)                             
 dplyr          0.7.8     2018-11-10 [1] CRAN (R 3.5.2)                             
 DT           * 0.5       2018-11-05 [1] CRAN (R 3.5.2)                             
 foreach      * 1.4.4     2017-12-12 [1] CRAN (R 3.5.2)                             
 fs             1.2.6     2018-08-23 [1] CRAN (R 3.5.2)                             
 ggplot2        3.1.0     2018-10-25 [1] CRAN (R 3.5.2)                             
 glue           1.3.0     2018-07-17 [1] CRAN (R 3.5.2)                             
 googleAuthR  * 0.7.0     2018-11-16 [1] CRAN (R 3.5.2)                             
 googleID     * 0.0.9001  2019-02-14 [1] Github (MarkEdmondson1234/googleID@b1edf53)
 gtable         0.2.0     2016-02-26 [1] CRAN (R 3.5.2)                             
 htmltools      0.3.6     2017-04-28 [1] CRAN (R 3.5.2)                             
 htmlwidgets  * 1.3       2018-09-30 [1] CRAN (R 3.5.2)                             
 httpuv         1.4.5.1   2018-12-18 [1] CRAN (R 3.5.2)                             
 httr           1.4.0     2018-12-11 [1] CRAN (R 3.5.2)                             
 iterators      1.0.10    2018-07-13 [1] CRAN (R 3.5.2)                             
 jsonlite       1.6       2018-12-07 [1] CRAN (R 3.5.2)                             
 later          0.8.0     2019-02-11 [1] CRAN (R 3.5.2)                             
 lazyeval       0.2.1     2017-10-29 [1] CRAN (R 3.5.2)                             
 magrittr       1.5       2014-11-22 [1] CRAN (R 3.5.2)                             
 memoise        1.1.0     2017-04-21 [1] CRAN (R 3.5.2)                             
 mime           0.6       2018-10-05 [1] CRAN (R 3.5.2)                             
 munsell        0.5.0     2018-06-12 [1] CRAN (R 3.5.2)                             
 openssl        1.2.1     2019-01-17 [1] CRAN (R 3.5.2)                             
 packrat        0.5.0     2018-11-14 [1] CRAN (R 3.5.2)                             
 pillar         1.3.1     2018-12-15 [1] CRAN (R 3.5.2)                             
 pkgbuild       1.0.2     2018-10-16 [1] CRAN (R 3.5.2)                             
 pkgconfig      2.0.2     2018-08-16 [1] CRAN (R 3.5.2)                             
 pkgload        1.0.2     2018-10-29 [1] CRAN (R 3.5.2)                             
 plyr         * 1.8.4     2016-06-08 [1] CRAN (R 3.5.2)                             
 prettyunits    1.0.2     2015-07-13 [1] CRAN (R 3.5.2)                             
 pROC         * 1.13.0    2018-09-24 [1] CRAN (R 3.5.2)                             
 processx       3.2.1     2018-12-05 [1] CRAN (R 3.5.2)                             
 promises       1.0.1     2018-04-13 [1] CRAN (R 3.5.2)                             
 ps             1.3.0     2018-12-21 [1] CRAN (R 3.5.2)                             
 purrr          0.3.0     2019-01-27 [1] CRAN (R 3.5.2)                             
 R6             2.3.0     2018-10-04 [1] CRAN (R 3.5.2)                             
 Rcpp           1.0.0     2018-11-07 [1] CRAN (R 3.5.2)                             
 RCurl          1.95-4.11 2018-07-15 [1] CRAN (R 3.5.2)                             
 rcytoscapejs * 0.0.7     2019-02-14 [1] local                                      
 remotes        2.0.2     2018-10-30 [1] CRAN (R 3.5.2)                             
 rlang          0.3.1     2019-01-08 [1] CRAN (R 3.5.2)                             
 rprojroot      1.3-2     2018-01-03 [1] CRAN (R 3.5.2)                             
 rsconnect      0.8.13    2019-01-10 [1] CRAN (R 3.5.2)                             
 rstudioapi     0.9.0     2019-01-09 [1] CRAN (R 3.5.2)                             
 scales         1.0.0     2018-08-09 [1] CRAN (R 3.5.2)                             
 sessioninfo    1.1.1     2018-11-05 [1] CRAN (R 3.5.2)                             
 shiny        * 1.2.0     2018-11-02 [1] CRAN (R 3.5.2)                             
 shinyjs      * 1.0       2018-01-08 [1] CRAN (R 3.5.2)                             
 shinythemes  * 1.1.2     2018-11-06 [1] CRAN (R 3.5.2)                             
 tibble         2.0.1     2019-01-12 [1] CRAN (R 3.5.2)                             
 tidyr        * 0.8.2     2018-10-28 [1] CRAN (R 3.5.2)                             
 tidyselect     0.2.5     2018-10-11 [1] CRAN (R 3.5.2)                             
 usethis      * 1.4.0     2018-08-14 [1] CRAN (R 3.5.2)                             
 withr          2.1.2     2018-03-15 [1] CRAN (R 3.5.2)                             
 xtable         1.8-3     2018-08-29 [1] CRAN (R 3.5.2)                             
 yaml           2.2.0     2018-07-25 [1] CRAN (R 3.5.2)                             

[1] /home/alp/R/x86_64-pc-linux-gnu-library/3.5
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library

**Comment by cannin** [comment]
Are you using rcytoscapejs or cyShiny? The log says rcytoscapejs. The underlying code is completely different. We decided to deprecate rcytoscapejs in favor of cyShiny because of cytoscapejs changes that made it hard to update to newer versions in rcytoscapejs. At this point rcytoscapejs makes use of a very old version of cytoscapejs; we redirect people to cyShiny to make them aware of the problem and that cyShiny is the new solution. But this likely prevents packrat from working properly in retrieving a package that no longer exists. 

You might not be in a position to migrate to cyShiny, therefore the last working rcytoscapejs exists here: https://github.com/cytoscape/cyjShiny/releases but telling rsconnect to use that zip file is a question best asked to the rsconnect people.

**Comment by alptaciroglu** [comment]
Oh that actually explains quite a bit. I started writing that app quite a while ago and did not make any changes to rcytoscapejs part. So I am using the old rcytoscapejs part. 

Though, I installed from zip file in the ->  https://github.com/cytoscape/cyjShiny.git
and was able to call it like old ways -> library(rcytoscapejs). Guess this is the intended way for old users. 

But anyways, If I conjure up enough time to migrate to newer version of the code, would rsconnect work without this issue?
